pkg/reaper/datacenter.go

```diff
@@ -52,20 +53,51 @@ func addInitContainer(reaperTemplate *reaperapi.ReaperClusterTemplate, dcConfig
 				Name: "REAPER_JMX_PASSWORD",
 				ValueFrom: &corev1.EnvVarSource{
 					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{Name: jmxUserSecretRef},
+						LocalObjectReference: jmxUserSecretRef,
 						Key:                  "password",
 					},
 				},
 			},
 		},
-		Args: []string{
-			"/bin/sh",
-			"-c",
-			"echo \"$REAPER_JMX_USERNAME $REAPER_JMX_PASSWORD\" > /config/jmxremote.password",
-		},
 		VolumeMounts: []corev1.VolumeMount{{
 			Name:      "server-config",
 			MountPath: "/config",
 		}},
-	})
+	}
+	if authEnabled {
+		// Note: dcConfig.Auth.SuperuserSecretRef should never be empty here
+		jmxInitContainer.Env = append(
+			jmxInitContainer.Env,
+			corev1.EnvVar{
+				Name: "SUPERUSER_JMX_USERNAME",
```

Comment:
This reflects K8ssandra 1.x but I don't really understand why we need to add those env vars. Reaper itself is not going to use them to open JMX connections, so they are only useful imo if a human attempts to open a JMX connection to the node using the superuser credentials. Am I missing something?

Reply:
We decided it would be best to reuse the superuser credentials for JMX. If/when we address https://github.com/k8ssandra/k8ssandra/issues/323, this would go away.

Reply:
Thank you for the pointer. Indeed when that ticket is implemented, none of these env vars would be required anymore.

However even when using the "old-style" JMX authentication through a `jmxremote.password` file, I still don't see why we need `SUPERUSER_JMX_USERNAME` and `SUPERUSER_JMX_PASSWORD`:

Indeed, Reaper itself is going to use `REAPER_JMX_USERNAME` and `REAPER_JMX_PASSWORD` to authenticate its JMX connections. These vars are always created and written to `jmxremote.password`, even when auth is disabled.

As far as I can tell, no component is making any use of the 2 extra vars that get created only when auth is enabled: `SUPERUSER_JMX_USERNAME` and `SUPERUSER_JMX_PASSWORD`. So I'm wondering if these 2 are necessary at all.

I will do a local test to verify this.

Reply:
You are correct about their not being used. Now I remember how we got here :) Initially, JMX auth was only turned on when Reaper was enabled. Then it was decided that it should be enabled regardless, but I believe a different set of credentials were generated. Users assumed that the superuser credentials should be used for `nodetool`. This caused some confusion. It was then decided to reuse the superuser credentials for JMX to make using `nodetool` less confusing.

Reply:
Ah, thanks for the explanation, now it makes a lot more sense. 

I now have a doubt: any call to nodetool in an e2e test should fail, and yet, the tests are passing... I need to investigate a bit further.

Reply:
OK confirmed:

1. The tests are passing because the current code has a flaw: auth is only being enabled on Cassandra pods when Reaper is also enabled. I need to fix that.
2. If I call `nodetool status` in a test where Reaper is enabled, I do get:

```
error: Authentication failed! Credentials required
-- StackTrace --
java.lang.SecurityException: Authentication failed! Credentials required
	at java.management/com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticationFailure(Unknown Source)
	at java.management/com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticate(Unknown Source)
[...]
command terminated with exit code 2
```

This needs to be fixed in test code by passing the appropriate credentials to `kubectl exec`. I will fix that too.

Reply:
I now have successfully decoupled the local vs remote JMX logic from the authentication logic.

In summary:

* If auth is enabled globally, all JMX clients must authenticate, including clients connecting to localhost.
* If auth is disabled globally, all JMX connections are unauthenticated, including remote ones. 
* If Reaper is not present, only local JMX connections are allowed.
* If Reaper is present, local and remote JMX connections are allowed. 

@jsanda does that look reasonable to you?

Reply:
Sounds good